clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -637,6 +638,9 @@ private void handleResponse(RequestHeader header, Struct body, long now) {
                 log.trace("Ignoring empty metadata response with correlation id {}.", header.correlationId());
                 this.metadata.failedUpdate(now);
             }
+            for (String topic : response.errorTopics(Errors.UNKNOWN_TOPIC_OR_PARTITION))
+                this.metadata.maybeSetExpiry(topic, now);
```

Comment:
Have removed the timestamp from the handling of UNKNOWN_TOPIC_OR_PARTITION. But timestamp is required for periodic expiry to ensure that a topic is only expired after the metadata expiry interval elapses even if there is a metadata update in the meantime triggered by a send to a new topic. Hence I have left the constants without replacing with enum,